services/bettersocial/fcmToken/sendChatNotificationByChannelMembers.js

```diff
@@ -0,0 +1,22 @@
+const sendChatNotification = require('./sendChatNotification');
+
+const sendChatNotificationByChannelMembers = async (
+  channelMembers = [],
+  payload = {},
+  options = {}
+) => {
+  const deduplicatedChannelMembers = [...new Set(channelMembers)];
+
+  deduplicatedChannelMembers.forEach((member) => {
+    const newPayload = {
+      ...payload,
+      data: {
+        ...payload.data,
+        is_annoymous: member.is_anonymous.toString()
+      }
+    };
+    sendChatNotification(member?.user_id, newPayload, options);
+  });
+};
+
+module.exports = sendChatNotificationByChannelMembers;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `sendChatNotificationByChannelMembers` is asynchronous but it doesn't use any `await` keyword inside. This could lead to unexpected behavior if the `sendChatNotification` function is asynchronous and returns a Promise, because these promises are not being awaited.

Also, there's no error handling in case `sendChatNotification` fails for some users. It would be better to add try-catch blocks around `sendChatNotification` calls or handle promise rejections.

Here's a suggested fix:

```diff
  const sendChatNotificationByChannelMembers = async (
    channelMembers = [],
    payload = {},
    options = {}
  ) => {
    const deduplicatedChannelMembers = [...new Set(channelMembers)];

+   const notificationsPromises = deduplicatedChannelMembers.map((member) => {
      const newPayload = {
        ...payload,
        data: {
          ...payload.data,
          is_annoymous: member.is_anonymous.toString()
        }
      };
-     sendChatNotification(member?.user_id, newPayload, options);
+     return sendChatNotification(member?.user_id, newPayload, options).catch((error) => {
+       console.error(`Failed to send notification to user ${member?.user_id}:`, error);
+     });
+   });

+   await Promise.all(notificationsPromises);
  };

  module.exports = sendChatNotificationByChannelMembers;
```
This change will ensure all notifications are sent concurrently and errors are logged without stopping the entire process.


<!-- This is an auto-generated comment by OSS CodeRabbit -->